File: api-gateway-microservice/src/main/scala/services/Application.scala
package services

import org.springframework.boot.SpringApplication
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.cloud.netflix.sidecar.EnableSidecar

/**
  * @author 刘佳兴
  * @version V1.0
  */

object Application {
  def main(args: Array[String]) {
    SpringApplication.run(Application)
  }
}

@SpringBootApplication
@EnableSidecar
class Application {

}

File: api-gateway-microservice/src/main/java/services/Application.java
package services;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.boot.autoconfigure.security.oauth2.client.EnableOAuth2Sso;
import org.springframework.cloud.netflix.feign.FeignClient;
import org.springframework.cloud.netflix.sidecar.EnableSidecar;
import org.springframework.cloud.netflix.zuul.EnableZuulProxy;
import org.springframework.security.oauth2.config.annotation.web.configuration.EnableAuthorizationServer;
import org.springframework.security.oauth2.config.annotation.web.configuration.EnableResourceServer;

/**
 * @author 刘佳兴
 */
@SpringBootApplication
@EnableSidecar
public class Application {
    public static void main(String[] args) {
        SpringApplication.run(Application.class, args);
    }
}
